### PR TITLE
Check blog token when computing is_registered

### LINKED ISSUE #16056: $manager->is_registered() method returns incorrect value
In the Jetpack connection flow, a site is "registered" if it has a blog token but no user token.

However, in the Jetpack\Connection\Manager class, the is_registered() method is implemented like this:

```php
	/**
	 * Returns true if the site has both a token and a blog id, which indicates a site has been registered.
	 *
	 * @access public
	 *
	 * @return bool
	 */
	public function is_registered() {
		$blog_id   = \Jetpack_Options::get_option( 'id' );
		$has_token = $this->is_active();
		return $blog_id && $has_token;
	}
```

This is incorrect, both in the description and implementation. It checks the user token (via `$this->is_active()`) rather than the blog token. It should be just this:

```php
	/**
	 * Returns true if the site has both a blog token and a blog id, which indicates a site has been registered.
	 *
	 * @access public
	 *
	 * @return bool
	 */
	public function is_registered() {
		$blog_id   = \Jetpack_Options::get_option( 'id' );
		$has_token = $this->get_access_token( false ); // fetches blog token
		return $blog_id && $has_token;
	}
```

I don't know all the implications of changing the behaviour of this function at this point, but as someone consuming the API it's definitely making life more difficult.

### PR DESCRIPTION
**Pull request by gravityrail**
Fixes #16056

#### Changes proposed in this Pull Request:

* Check blog token when computing $manager->is_registered()

#### Testing instructions:
* Register, but do not activate a site
* `(new Connection_Manager())->is_registered()` should return true
* `(new Connection_Manager())->is_active()` should return false

### CODE REVIEW

File: packages/connection/src/class-manager.php [hunk 1]
```diff
@@ -519,9 +519,9 @@ public function is_active() {
 	 * @return bool
 	 */
 	public function is_registered() {
-		$blog_id   = \Jetpack_Options::get_option( 'id' );
-		$has_token = $this->is_active();
-		return $blog_id && $has_token;
+		$has_blog_id    = ! ! \Jetpack_Options::get_option( 'id' );
+		$has_blog_token = ! ! $this->get_access_token( false );
```
Comment (jeherve) [inline packages/connection/src/class-manager.php:523]: That may be a stupid question, but is there a reason to use `!!` over `(bool)` in scenarios like this one? I find the latter more readable, but maybe I'm missing something here.

Comment (gravityrail) [reply packages/connection/src/class-manager.php:523]: Fair point, `(bool)` is probably better.

**Comment by leogermani** [comment]
Thanks for this PR,

I'll cc @sergeymitr @kbrown9 and @dereksmart here.

I think we just need to make sure this change doesn't break anything.

A quick search shows me that there's only this place using it:

https://github.com/Automattic/jetpack/blob/master/_inc/class.jetpack-provision.php#L51

And the API, as @gravityrail already pointed out. But we want that fixed.

So, who else could be consuming that API?

Also, it would be nice to add some tests to this, right?

**Comment by kbrown9** [comment]
> A quick search shows me that there's only this place using it:
> 
> https://github.com/Automattic/jetpack/blob/master/_inc/class.jetpack-provision.php#L51
> 

It's looks like `Jetpack_Provision::register_and_build_request_body` is meant to be called when setting up a new site. However, it might be possible for this method to be called on a site that's registered but not authorized. If it is possible, we need to make sure that scenario is handled properly.

> And the API, as @gravityrail already pointed out. But we want that fixed.
> 
> So, who else could be consuming that API?
 
We definitely need to make sure that any current users of that API are ready for this change. I have no idea who might be using it though. 

> Also, it would be nice to add some tests to this, right?

I agree; some unit test coverage would be good.

**Comment by gravityrail** [comment]
I strongly suspect we (Automattic) are the only consumers of the API. I think it's reasonable to act accordingly.

**Comment by leogermani** [comment]
@kbrown9 thanks for digging into this.

No objection. I'm adding WorDBless in #16194 and adjusting it along the way. So I can come back here and write some tests later.

I've created an issue so we don't forget!

**Pull request merged**